src/jit/emit.cpp

```diff
@@ -5475,7 +5475,7 @@ void emitter::emitOutputDataSec(dataSecDsc* sec, BYTE* dst)
 #ifdef _TARGET_ARM_
                 target = (BYTE*)((size_t)target | 1); // Or in thumb bit
 #endif
-                bDst[i] = target;
+                bDst[i] = (TargetPointerType)target;
```

Comment:
Why not change the `BYTE* target` definition to `TargetPointerType* target`? And then update the line above to:
```
target = (TargetPointerType*)((TargetPointerType)target | 1); // Or in thumb bit
```

Reply:
I'll try to do it this way. Thanks

Reply:
@BruceForstall `target` can also hold 64-bit addresses here